processing/src/main/java/org/apache/carbondata/processing/surrogatekeysgenerator/csvbased/CarbonCSVBasedSeqGenStep.java

```diff
@@ -853,8 +873,10 @@ private String getCarbonLocalBaseStoreLocation() {
 
   private Object[] process(Object[] r) throws RuntimeException {
     try {
+      Object[] actualRow = new Object[r.length];
+      System.arraycopy(r, 0, actualRow, 0, r.length);
```

Comment:
Can't we avoid this object[] instance and arraycopy? 
I guess we can merge `changeNullValueToNullString` and `populateOutputRow` and avoid arraycopy.